### ISSUE TITLE
[Feature request] Automatically stringify JSON

### ISSUE THREAD

**Issue by KaKi87** [mpetrunic/fastify-sse-v2#80]
Thanks

**Comment by millette** [comment]
@mpetrunic  If there's interest, I can make a PR for this.

**Comment by mpetrunic** [comment]
Maybe you can first propose approach here (without breaking changes)? :)

**Comment by KaKi87** [comment]
Simple : `reply.sse({ data: JSON.stringify({ 'foo': 'bar' }) })` keeps working but can be simplified with `reply.sse({ data: { 'foo': 'bar' } })`, which currently has no use case since it sends `[Object object]`.

Thanks

**Comment by mpetrunic** [comment]
Ah I see, stringify data if it's an object. Yeah, I'm happy to accept PR with the changes.

**Comment by millette** [comment]
Oh wow, I completely missed this issue. I'll have a look around tomorrow and get to it if nobody else started the PR.

**Comment by daimalou** [comment]
In some cases, `JSON.stringify` is not very good, and we need other libraries to improve stringify performance.

**Comment by daimalou** [comment]
SSE common scenarios involve high-frequency requests. We need to reduce consumption.

**Comment by KaKi87** [comment]
Yeah, I was just making an example, in practice I guess the library that Fastify currently uses for serializing JSON should be used here.

**Comment by millette** [comment]
@KaKi87 Although it's old, fastify uses https://www.npmjs.com/package/JSONStream#jsonstreamstringifyobjectopen-sep-close

**Comment by daimalou** [comment]
just use JSON.stringify, we can edit this file.

https://github.com/mpetrunic/fastify-sse-v2/blob/master/src/sse.ts



```ts
export function isObject(value: any) {
  return typeof value === 'object' && value !== null;
}

...

  if (chunk.data) {
    if (isObject(chunk.data)) {
      payload += `data: ${JSON.stringify(chunk.data)}\n`;
    } else {
      payload += `data: ${chunk.data}\n`;
    }
  }
```
